File: src/main/java/bot/farm/steam_news_bot/entity/UserGameState.java
package bot.farm.steam_news_bot.entity;

import java.util.Objects;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Entity class representing the game state of a user.
 * Provides properties and methods to store and manipulate user game state data.
 */
@Setter
@Getter
@NoArgsConstructor
@Entity
public class UserGameState {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id", nullable = false)
  private Long id;
  
  @ManyToOne()
  @JoinColumn(name = "user_id")
  private User user;
  
  @ManyToOne(
      cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
  @JoinColumn(name = "game_id")
  private Game game;
  
  private boolean isWished;
  private boolean isBanned;
  private boolean isOwned;
  
  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    UserGameState that = (UserGameState) o;
    return Objects.equals(id, that.id)
        && Objects.equals(user, that.user) && Objects.equals(game, that.game);
  }
  
  @Override
  public int hashCode() {
    return Objects.hash(id, user, game);
  }
}

File: src/main/kotlin/domain/state/entity/UserGameState.kt
package sidim.doma.domain.state.entity

import org.jetbrains.exposed.sql.Table
import sidim.doma.domain.game.entity.Games
import sidim.doma.domain.user.entity.Users

data class UserGameState(
    val userId: String,
    val gameId: String,
    val isWished: Boolean,
    val isBanned: Boolean,
    val isOwned: Boolean
)

object UserGameStates : Table("user_game_state") {
    val id = long("id").autoIncrement()
    val userId = varchar("user_id", 255).references(Users.chatId)
    val gameId = varchar("game_id", 255).references(Games.appid)
    val isWished = bool("is_wished")
    val isBanned = bool("is_banned")
    val isOwned = bool("is_owned")

    override val primaryKey = PrimaryKey(id)
}